master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/AbstractMetaManager.java

```diff
@@ -436,6 +446,12 @@ public void updateWorkerEventMeta(int workerEventTypeValue, List<WorkerInfo> wor
     }
   }
 
+  public void updateMetaByReportWorkerDecommission(List<WorkerInfo> workers) {
+    synchronized (this.workers) {
+      decommissionWorkers.addAll(workers);
```

Comment:
Before this, worker decommission will ReportWorkerUnavailable, which indicates the worker would be shutdown, then Client would quickly tell worker 
commits those associated partitions,  If we change shutdownWorkers to decommissionWorkers, May be decommission take longer time than before.

Reply:
It seems that we should return decommissionWorkers to the client through heartbeat, but do we need to distinguish between shutdown and decommission on the client side?

Reply:
> It seems that we should return decommissionWorkers to the client through heartbeat, but do we need to distinguish between shutdown and decommission on the client side?

Currently, the client doesn't need to know that.

Reply:
Just. return decommission worker + shutdown worker to client side?

Reply:
> Just. return decommission worker + shutdown worker to client side?

I think so too

Reply:
Looks good!